### ISSUE TITLE
Last update (v.2.5.908) broke something in my project - Pluralization/Singularization

### ISSUE THREAD

**Issue by prbarbosa** [ErikEJ/EFCorePowerTools#1290]
Hi,

I've been using this EFCorePowerTools for the last 5 months. 

Everything was working fine until this last update v2.5.908.

I'm using EF Core 3 (3.1.22)

Part of the code before the change (project working like this)

        public virtual DbSet<Customer> Customers { get; set; } //type ok, variable name plural, table name in SQL is Customer 
        public virtual DbSet<Delivery> Deliveries { get; set; } //type ok, variable name plural, table name in SQL is Delivery
        public virtual DbSet<GradeType> GradeTypes { get; set; } //type ok, variable name plural, table name in SQL is GradeType
        public virtual DbSet<RawMaterial> RawMaterials { get; set; } //table name in SQL is RawMaterials (yes, plural)
        public virtual DbSet<SiteLocation> SiteLocations { get; set; } //table name in SQL is SiteLocation
        public virtual DbSet<StockLocation> StockLocations { get; set; } //table name in SQL is StockLocation  

After the change

        public virtual DbSet<Customer> Customer { get; set; } //type ok but variable name is now singular
        public virtual DbSet<Delivery> Delivery { get; set; } //type ok but variable name is now singular
        public virtual DbSet<GradeType> GradeType { get; set; } //type ok but variable name is now singular
        public virtual DbSet<**RawMaterials**> RawMaterials { get; set; } //type changed, but variable ok (that's the table name, RawMaterials)
        public virtual DbSet<SiteLocation> SiteLocation { get; set; } //type ok but variable name is now singular
        public virtual DbSet<StockLocation> StockLocation { get; set; } //type ok but variable name is now singular

This happens in a couple of other cases, breaking my project. 

My question is, is this a bug or a feature? If it's a bug, I won't change anything on my code and will rollback to previous version. If it's a feature, then I have to adapt.

Note: The option to pluralize or singularize generated object names doesn't seem to do any changes regarding this.

**Comment by ErikEJ** [comment]
No change as far as I know. Are you using the "use Database names" option?

**Comment by prbarbosa** [comment]
I can try that, but no... I'm not using the "user Database names" option. I didn't change any settings,

**Comment by ErikEJ** [comment]
I was able to repro with 3.1 - it looks like pluralization is off. I will investigate asap

**Comment by ErikEJ** [comment]
Latest daily is .915. Guess you were too quick.

**Comment by prbarbosa** [comment]
@ErikEJ 

I can see that the files are now properly generated.

Thank you for fixing it.

---

I noticed a couple of changes in the generated files (i'm comparing with previous versions). I guess this is all normal.

**Comment by ErikEJ** [comment]
What changes??

**Comment by prbarbosa** [comment]
Well, for example, at the end of the context file there's a call to a function 

OnModelCreatingGeneratedProcedures(modelBuilder);

This is just before calling OnModelCreatingPartial(modelBuilder);

On my previous version of the generated context file, the call to OnModelCreatingGeneratedProcedures doesn't exist, only on the new file.

I am using stored procedures.

**Comment by ErikEJ** [comment]
Yes, that is a change as described in the release notes for the latest daily.

**Comment by prbarbosa** [comment]
Ok.

Thank you again for fixing this issue.

Cheers

**Comment by kateract** [comment]
I am also able to confirm that we were experiencing the same issue in 908, and that the problem has been resolved with the latest build (915).  Is there a timeline for when this will be published?

**Comment by ErikEJ** [comment]
@kateract within some days. Why do you need it published?

**Comment by ErikEJ** [comment]
@kateract Published to VS Marketplace now
